mlflow/store/artifact/utils/models.py

```diff
@@ -63,6 +64,17 @@ def _parse_model_uri(uri):
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
     path = parsed.path
+    if parsed.fragment:
```

Comment:
I *think* fragment occurs after query in all cases. We should double check. If that's true, this logic should occur after the parsed.query logic.

Reply:
This was totally broken. Great catch. Added tests to check for ordering of both and adjusted the logic to catch ordering based on substring index matching.